zerver/tests/test_upload.py

```diff
@@ -221,9 +222,26 @@ def test_file_upload_authed(self) -> None:
         self.assertEqual(result.status_code, 401)
 
     def test_serve_local_file_unauthed_invalid_token(self) -> None:
-        result = self.client_get('/user_uploads/temporary/badtoken')
+        result = self.client_get('/user_uploads/temporary/badtoken/file.png')
         self.assert_json_error(result, "Invalid token")
 
+    def test_serve_local_file_unauthed_altered_filename(self) -> None:
+        self.login('hamlet')
+        fp = StringIO("zulip!")
+        fp.name = "zulip.txt"
+        result = self.client_post("/json/user_uploads", {'file': fp})
+        url = '/json' + result.json()["uri"]
+
+        result = self.client_get(url)
+        self.assert_json_success(result)
+        data = result.json()
+        url_only_url = data['url']
+
+        self.assertTrue(url_only_url.endswith('zulip.txt'))
+        url_only_url_changed_filename = url_only_url.strip('zulip.txt') + 'differentname.exe'
```

Comment:
That isn’t what `.strip` means.

```pycon
>>> 'pillowslip'.strip('zulip.txt')
'ows'
```

Reply:
Yeah, probably `os.path.dirname`  or `.rsplit("/", 2)[0]` or similar.

Reply:
Oh yeah that was dumb, `split` was what I wanted, fixed